backend/src/main/java/de/iav/backend/controller/TimeSlotController.java

```diff
@@ -0,0 +1,81 @@
+package de.iav.backend.controller;
+
+import de.iav.backend.model.TimeSlotResponseDTO;
+import de.iav.backend.model.TimeSlotWithoutIdDTO;
+import de.iav.backend.service.TimeSlotService;
+import jakarta.validation.Valid;
+import java.util.List;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+
+@RestController
+@RequestMapping("/api/toastMasterManager/timeslots")
+public class TimeSlotController {
+
+    private final TimeSlotService timeSlotService;
+
+
+    public TimeSlotController(TimeSlotService timeSlotService) {
+        this.timeSlotService = timeSlotService;
+    }
+
+    @GetMapping
+    public List<TimeSlotResponseDTO> getAllTimeSlots(            @RequestParam(required = false) String title,
+                                                                 @RequestParam(required = false) String description,
+                                                                 @RequestParam(required = false) String green,
+                                                                 @RequestParam(required = false) String amber,
+                                                                 @RequestParam(required = false) String red) {
+
+        if (title != null) {
+            return timeSlotService.getTimeSlotsByTitle(title);}
+        else if (description != null) {
+            return timeSlotService.getTimeSlotsByDescription(description);}
+        else if (green != null) {
+            return timeSlotService.getTimeSlotsByGreen(green);}
+        else if (amber != null) {
+            return timeSlotService.getTimeSlotsByAmber(amber);}
+        else if (red != null) {
+            return timeSlotService.getTimeSlotsByRed(red);}
+        return timeSlotService.getAllTimeSlots();
+    }
```

Comment:
Ich würde den Endpunkt aufteilen in 2 Endpunkte: 1x getAll und 1x einen der die Parameter entgegennimmt